net/redisclient.go

```diff
@@ -87,6 +101,96 @@ const (
 	defaultConnMetricsInterval = 60 * time.Second
 )
 
+// JumpHash
+// https://arxiv.org/pdf/1406.2294.pdf
+type JumpHash struct {
+	hash   hash.Hash64
+	shards []string
+}
+
+func NewJumpHash(shards []string) redis.ConsistentHash {
+	return &JumpHash{
+		hash:   fnv.New64(),
+		shards: shards,
+	}
+}
+
+func (j *JumpHash) Get(k string) string {
+	_, err := j.hash.Write([]byte(k))
+	if err != nil {
+		log.Fatalf("Failed to write %s to hash: %v", k, err)
+	}
+
+	key := j.hash.Sum64()
+	h := jump.Hash(key, len(j.shards))
+	return j.shards[int(h)%len(j.shards)]
+}
+
+// Multi-probe consistent hashing - mpchash
+// https://arxiv.org/pdf/1505.00062.pdf
+type multiprobe struct {
+	hash   *mpchash.Multi
+	shards []string
+}
+
+func NewMultiprobe(shards []string) redis.ConsistentHash {
+	return &multiprobe{
+		// 2 seeds and k=21
+		hash: mpchash.New(shards, siphash64seed, [2]uint64{1, 2}, 21),
+		// 2 seeds and k=41
+		//hash:   mpchash.New(shards, siphash64seed, [2]uint64{1, 2}, 41),
+		shards: shards,
+	}
+}
+
+func (mc *multiprobe) Get(k string) string {
+	return mc.hash.Hash(k)
+}
+func siphash64seed(b []byte, s uint64) uint64 { return siphash.Hash(s, 0, b) }
+
+// rendezvous copied from github.com/go-redis/redis/v8@v8.3.3/ring.go
+type rendezvousWrapper struct {
+	*rendezvous.Rendezvous
+}
+
+func (w rendezvousWrapper) Get(key string) string {
+	return w.Lookup(key)
+}
+
+func NewRendezvous(shards []string) redis.ConsistentHash {
+	return rendezvousWrapper{rendezvous.New(shards, xxhash.Sum64String)}
+}
+
+// rendezvous vnodes
+type rendezvousVnodes struct {
+	*rendezvous.Rendezvous
+	table map[string]string
+}
+
+func (w rendezvousVnodes) Get(key string) string {
+	k := w.Lookup(key)
+	v, ok := w.table[k]
+	if !ok {
+		log.Printf("not found: %s in table for input: %s, so return %s", k, key, v)
+	}
+	return v
+}
+
+func NewRendezvousVnodes(shards []string) redis.ConsistentHash {
+	N := 100
+	vshards := make([]string, N*len(shards))
+	table := make(map[string]string)
+	for i := 0; i < N; i++ {
+		for j, shard := range shards {
+			//vshard := fmt.Sprintf("%d%s", i, shard) // prefix
```

Comment:
Lets remove this commented out

Reply:
we should decide if we want to provide prefix or suffix vshards

Reply:
Assuming xxhash has good https://en.wikipedia.org/wiki/Avalanche_effect it should not matter which one is used so lets remove the comment and stick with suffix

Reply:
removed